modules/dag-l1/src/main/scala/org/tessellation/dag/l1/Main.scala

```diff
@@ -46,19 +48,22 @@ object Main extends TessellationIOApp[Run]("", "DAG L1 node", version = BuildInf
           .start(storages, services, queues, p2pClient, rumorHandler, nodeId, cfg)
           .asResource
 
-        api = HttpApi.make[IO](storages, queues, keyPair.getPrivate, services, sdkPrograms, sdk.nodeId)
+        api = HttpApi.make[IO](storages, queues, keyPair.getPrivate, services, programs, sdk.nodeId)
         _ <- MkHttpServer[IO].newEmber(ServerName("public"), cfg.http.publicHttp, api.publicApp)
         _ <- MkHttpServer[IO].newEmber(ServerName("p2p"), cfg.http.p2pHttp, api.p2pApp)
         _ <- MkHttpServer[IO].newEmber(ServerName("cli"), cfg.http.cliHttp, api.cliApp)
         stateChannel = new StateChannel(cfg, keyPair, p2pClient, queues, nodeId, services, storages, validators)
         _ <- {
           method match {
-            case _: RunInitialValidator =>
-              storages.node.tryModifyState(NodeState.Initial, NodeState.ReadyToJoin) >>
-                services.session.createSession
+            case cfg: RunInitialValidator =>
+              programs.l0PeerDiscovery.discoverFrom(cfg.l0Peer) >>
+                storages.node.tryModifyState(NodeState.Initial, NodeState.ReadyToJoin) >>
```

Comment:
we may think about it as a `Genesis` for later usage, when the initial peer will make a download to pre-fetch latest global snapshot